### PR TITLE
Add R_t Indicator Data

### PR DESCRIPTION
**Pull request by erccarls**
Generate R_t inference based on mortality and case data when available (hospital data is very noisy so additional signal processing work is needed if we care. We probably don't care unless the data gets much better).  

We also add an end-to-end test for Montana to capture some notion of test coverage.

This PR addresses issue #<insert number>

### Please Check
- [ ] Are the [JSON schemas](https://github.com/covid-projections/covid-data-model/tree/master/api/schemas) updated?
- [ ] Are the [api docs](https://github.com/covid-projections/covid-data-model/blob/master/api/README.V1.md) updated?
- [ ] Are tests passing?

### CODE REVIEW

File: pyseir/inference/infer_rt.py [hunk 1]
```diff
@@ -324,10 +325,13 @@ def infer_all(self, plot=False, shift_deaths=0):
...
 
-        if len(self.get_timeseries(TimeseriesType.NEW_CASES)) > 0:
+        cases = self.get_timeseries(TimeseriesType.NEW_CASES.value)[2]
+        deaths = self.get_timeseries(TimeseriesType.NEW_DEATHS.value)[2]
+
+        if np.sum(cases) > 5:
             available_timeseries.append(TimeseriesType.NEW_CASES)
 
-        if len(self.get_timeseries(TimeseriesType.NEW_DEATHS)) > 0:
+        if np.sum(deaths) > 5:
```
Comment (ghop02) [inline pyseir/inference/infer_rt.py:334]: Why 5?

File: pyseir/inference/infer_rt.py [hunk 2]
```diff
@@ -37,7 +38,7 @@ class RtInferenceEngine:
     ref_date:
         Reference date to compute from.
     confidence_intervals: list(float)
-        Confidence interval to compute. 0.90 would be 90% credible
+        Confidence interval to compute. 0.95 would be 90% credible
```
Comment (ghop02) [inline pyseir/inference/infer_rt.py:41]: 95%

File: pyseir/cli.py [hunk 3]
```diff
@@ -116,7 +116,7 @@ def _map_outputs(state=None, output_interval_days=4, states_only=False,
                          run_mode=run_mode, output_dir=output_dir)
 
 
-def _run_all(state=None, run_mode=DEFAULT_RUN_MODE, generate_reports=True, output_interval_days=4,
+def _run_all(state=None, run_mode=DEFAULT_RUN_MODE, generate_reports=False, output_interval_days=4,
```
Comment (igorkofman) [inline pyseir/cli.py:119]: is this change intended?

File: pyseir/deployment/webui_data_adaptor_v1.py [hunk 4]
```diff
@@ -218,14 +218,34 @@ def map_fips(self, fips):
...
+            try:
+                rt_results = load_Rt_result(fips)
+                rt_results.index = rt_results['Rt_MAP_composite'].index.strftime('%m/%d/%y')
+
+                merged = output_model.merge(rt_results[['Rt_MAP_composite', 'Rt_ci95_composite']],
+                    right_index=True, left_on='date', how='left')
+
+                output_model[schema.RT_INDICATOR] = merged['Rt_MAP_composite']
+                # 90% confidence intervals run from 5% - 95%
+                output_model[schema.RT_INDICATOR_CI90] = merged['Rt_ci95_composite'] - merged['Rt_MAP_composite']
```
Comment (igorkofman) [inline pyseir/deployment/webui_data_adaptor_v1.py:234]: How should we interpret the CI90:  with 90% probability the value is between rt_indicator - ci90 to rt_indicator + ci90?

File: pyseir/inference/infer_rt.py [hunk 5]
```diff
@@ -324,10 +325,13 @@ def infer_all(self, plot=False, shift_deaths=0):
         df_all = None
         available_timeseries = []
 
-        if len(self.get_timeseries(TimeseriesType.NEW_CASES)) > 0:
+        cases = self.get_timeseries(TimeseriesType.NEW_CASES.value)[2]
```
[the remark this answers is not in the document]
Comment (igorkofman) [reply pyseir/inference/infer_rt.py:328]: +1

File: pyseir/inference/infer_rt.py (hunk 1 above)
Comment (igorkofman) [reply pyseir/inference/infer_rt.py:334]: +1, would love come comments on what's going on here

File: pyseir/cli.py (hunk 3 above)
Comment (erccarls) [reply pyseir/cli.py:119]: Yes, the default was outdated w.r.t. the CLI default.

File: pyseir/deployment/webui_data_adaptor_v1.py (hunk 4 above)
Comment (erccarls) [reply pyseir/deployment/webui_data_adaptor_v1.py:234]: That is correct. Comment added.

File: pyseir/inference/infer_rt.py (hunk 2 above)
Comment (erccarls) [reply pyseir/inference/infer_rt.py:41]: It is correct as is.  This is the upper bound to compute which provides a 90% confidence.

File: pyseir/inference/infer_rt.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (erccarls) [reply pyseir/inference/infer_rt.py]: Added. The docstring for this method also provides documentation.

File: pyseir/inference/infer_rt.py (hunk 1 above)
Comment (erccarls) [reply pyseir/inference/infer_rt.py]: Promoted to configurable instance variables and added comment.

File: pyseir/inference/infer_rt.py [hunk 6]
```diff
@@ -344,8 +348,11 @@ def infer_all(self, plot=False, shift_deaths=0):
                 df[f'Rt_MAP__{timeseries_type.value}'] = posteriors.idxmax()
                 for ci in self.confidence_intervals:
                     ci_low, ci_high = self.highest_density_interval(posteriors, ci=ci)
-                    df[f'Rt_ci{int(100 * (1 - ci / 2))}__{timeseries_type.value}'] = ci_low
-                    df[f'Rt_ci{int(100 * ci / 2)}__{timeseries_type.value}'] = ci_high
+
+                    low_val = 1 - ci
+                    high_val = ci
+                    df[f'Rt_ci{int(math.floor(100 * low_val))}__{timeseries_type.value}'] = ci_low
+                    df[f'Rt_ci{int(math.floor(100 * high_val))}__{timeseries_type.value}'] = ci_high
```
[the remark this answers is not in the document]
Comment (erccarls) [reply pyseir/inference/infer_rt.py:365]: It is a configurable parameter set in the constructor.

File: test/test_pyseir_end_to_end_test.py [hunk 7]
```diff
@@ -0,0 +1,20 @@
+import os
+import json
+import numpy as np
+from pyseir import cli
+from pyseir.utils import get_run_artifact_path, RunArtifact
+import libs.datasets.can_model_output_schema as schema
+
+
+def test__pyseir_end_to_end():
```
[the remark this answers is not in the document]
Comment (erccarls) [reply test/test_pyseir_end_to_end_test.py:9]: About 2 minutes I think.

[the remark this answers is not in the document]
Comment (erccarls) [reply test/test_pyseir_end_to_end_test.py:9]: I would have added components, but it is hard without mocks since there is dependency structure.  And mocks take a long time, which I sadly do not have much of :(
